### PR TITLE
[DOC] Add RTMDet doc coder and loss part

### PR DESCRIPTION
**Pull request by PeterH0323**
## Motivation

Add RTMDet doc coder and loss part.

## Modification

`docs/zh_cn/algorithm_descriptions/rtmdet_description.md`

### CODE REVIEW

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+> This coder encodes gt bboxes (x1, y1, x2, y2) into (top, bottom, left, right) and decode it back to the original.
+>
+> 这个编码器将 gt bboxes (x1, y1, x2, y2) 编码为 (top, bottom, left, right)，并且解码至原图像上
+
+MMDet 实现源码
+
+```python
+@TASK_UTILS.register_module()
+class DistancePointBBoxCoder(BaseBBoxCoder):
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:17]: 代码只贴核心就行，不需要整个类完全贴过来，会导致文档特别长

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+## Loss
+
+参与 Loss 计算的共有两个值：`loss_cls` 和 `loss_bbox`，其各自使用的 Loss 方法如下：
+
+- `loss_cls`：`mmdet.QualityFocalLoss`
+- `loss_bbox`：`mmdet.GIoULoss`
+
+### QualityFocalLoss
+
+Quality Focal Loss (QFL) 是 [Generalized Focal Loss: Learning Qualified and Distributed Bounding Boxes for Dense Object Detection](https://arxiv.org/abs/2006.04388) 的变体。
```
Comment (satuoqaq) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:85]: gfocal loss 包括qfl和dfl，这里只是一部分，用变体不太好吧

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+其可以将离散标签的 `focal loss` 泛化到连续标签上，从而直接预测分类分数和 IoU 分数的乘积。
```
Comment (satuoqaq) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:87]:
```suggestion
其可以将离散标签的 `focal loss` 泛化到连续标签上，将 bboxes 与 gt 的 IoU 的作为分类分数的标签，使得分类分数为表征回归质量的分数 。
```

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+参与 Loss 计算的共有两个值：`loss_cls` 和 `loss_bbox`，其各自使用的 Loss 方法如下：
+
+- `loss_cls`：`mmdet.QualityFocalLoss`
+- `loss_bbox`：`mmdet.GIoULoss`
+
+### QualityFocalLoss
+
+Quality Focal Loss (QFL) 是 [Generalized Focal Loss: Learning Qualified and Distributed Bounding Boxes for Dense Object Detection](https://arxiv.org/abs/2006.04388) 的一部分。
+
+其可以将离散标签的 `focal loss` 泛化到连续标签上，将 bboxes 与 gt 的 IoU 的作为分类分数的标签，使得分类分数为表征回归质量的分数。
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:65]: 这可以贴论文里面的公式原图

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        """
+        assert reduction_override in (None, 'none', 'mean', 'sum')
+        reduction = (
+            reduction_override if reduction_override else self.reduction)
+        if self.use_sigmoid:
+            if self.activated:
+                calculate_loss_func = quality_focal_loss_with_prob
+            else:
+                calculate_loss_func = quality_focal_loss
+            loss_cls = self.loss_weight * calculate_loss_func(
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:102]: 这些都不要，只要贴核心实现就行。

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            if pred.dim() == weight.dim() + 1:
+                weight = weight.unsqueeze(1)
+            return (pred * weight).sum()  # 0
+        assert reduction_override in (None, 'none', 'mean', 'sum')
+        reduction = (
+            reduction_override if reduction_override else self.reduction)
+        if weight is not None and weight.dim() > 1:
+            assert weight.shape == pred.shape
+            weight = weight.mean(-1)
+        loss = self.loss_weight * giou_loss(
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:150]: 只贴核心实现

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+>
+> 这个编码器将 gt bboxes (x1, y1, x2, y2) 编码为 (top, bottom, left, right)，并且解码至原图像上
+
+MMDet 编码的核心源码：
+
+```python
+def bbox2distance(points: Tensor, bbox: Tensor, ...) -> Tensor:
+    """
+        points (Tensor): 需要计算的点（该点已经乘上了 strides），Shape (n, 2) or (b, n, 2), [x, y].
+        bbox (Tensor): BBox 的 xyxy，Shape (n, 4) or (b, n, 4), "xyxy" format.
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:19]: bbox 为乘上 stride 的网络回归分支预测值

**Review by hhaAndroid — APPROVED** [review]
